src/sentry/services/hybrid_cloud/organization/impl.py

```diff
@@ -494,6 +500,54 @@ def delete_option(self, *, organization_id: int, key: str) -> None:
         orm_organization = Organization.objects.get_from_cache(id=organization_id)
         orm_organization.delete_option(key)
 
+    def get_team_memberships(
+        self, *, organization_id: int, team_ids: List[int]
+    ) -> List[RpcTeamMembership]:
+        members: Dict[int, RpcTeamMembership] = {}
+        for omt in RangeQuerySetWrapper(
+            OrganizationMemberTeam.objects.filter(team_id__in=team_ids).prefetch_related(
+                "organizationmember"
+            )
+        ):
+            if omt.organizationmember_id not in members:
+                members[omt.organizationmember_id] = RpcTeamMembership(
+                    user_id=omt.organizationmember.user_id,
+                    user_email=omt.organizationmember.get_email(),
+                    member_id=omt.organizationmember_id,
+                    team_ids=[],
+                )
+            members[omt.organizationmember_id].team_ids.append(omt.team_id)
+
+        return list(members.values())
+
+    def create_team(
+        self,
+        *,
+        organization_id: int,
+        slug: str,
+        name: Optional[str] = None,
+        idp_provisioned: bool = False,
+        creator_user_id: Optional[int] = None,
+    ) -> Optional[RpcTeam]:
+        try:
+            with transaction.atomic(router.db_for_write(Team)):
+                team = Team.objects.create(
+                    name=name or slug,
+                    slug=slug,
+                    idp_provisioned=idp_provisioned,
+                    organization_id=organization_id,
+                )
+
+            team_created.send_robust(
+                organization_id=organization_id,
+                user_id=creator_user_id,
+                team_id=team.id,
+                sender=None,
+            )
+            return serialize_rpc_team(team)
+        except (IntegrityError, MaxSnowflakeRetryError):
```

Comment:
I wonder why we are using snowflake ids for teams? Teams aren't generally in control silo :think

Reply:
I'll remove this from the except block.